internal/guest/storage/devicemapper/devicemapper.go

```diff
@@ -216,6 +219,43 @@ func makeTableIoctl(name string, targets []Target) *dmIoctl {
 	return d
 }
 
+// CreateDeviceWithRetryErrors keeps retrying to create device mapper target
+func CreateDeviceWithRetryErrors(
+	ctx context.Context,
+	name string,
+	flags CreateFlags,
+	targets []Target,
+	errs ...error,
+) (string, error) {
+retry:
+	for {
+		dmPath, err := CreateDevice(name, flags, targets)
+		if err == nil {
+			return dmPath, nil
+		}
+		log.G(ctx).WithError(err).Error("CreateDevice error")
+		// In some cases
+		dmErr, ok := err.(*dmError)
+		if !ok {
+			return "", err
+		}
+		// check retry-able errors
+		for _, e := range errs {
+			if errors.Is(dmErr.Err, e) {
```

Comment:
If a non-retriable error is encountered, does it make sense to fail fast?

Reply:
that's what it does, we loop through possible errors and in case we hit a non-retriable error, we finish the loop and return error on L255